packages/grid/_modules_/grid/hooks/features/export/seralizers/csvSeraliser.ts

```diff
@@ -15,37 +14,38 @@ const serialiseCellValue = (value) => {
 };
 
 export function serialiseRow(
-  row: GridRowModel,
+  id: GridRowId,
   columns: GridColumns,
   getCellValue: (id: GridRowId, field: string) => GridCellValue,
 ): Array<string> {
   const mappedRow: string[] = [];
   columns.forEach(
     (column) =>
       column.field !== gridCheckboxSelectionColDef.field &&
-      mappedRow.push(serialiseCellValue(getCellValue(row.id, column.field))),
+      mappedRow.push(serialiseCellValue(getCellValue(id, column.field))),
   );
   return mappedRow;
 }
 
 export function buildCSV(
   columns: GridColumns,
-  rows: GridRowModel[],
+  rowIds: GridRowId[],
   selectedRows: Record<GridRowId, boolean>,
   getCellValue: (id: GridRowId, field: string) => GridCellValue,
 ): string {
-  const selectedRowsIds = Object.keys(selectedRows);
+  let rowIdsToSerialize = [...rowIds];
+  const selectedRowIds = Object.keys(selectedRows);
 
-  if (selectedRowsIds.length) {
-    rows = rows.filter((row) => selectedRowsIds.includes(`${row.id}`));
+  if (selectedRowIds.length) {
```

Comment:
this logic should not be here. Is it a big refactoring? Maybe for another PR...

Reply:
Which logic? The one to compute the records for export or the cast of the id to string?

Reply:
BuildCsv should not care about which rows are selected. It should just receive rows and convert them to CSV. This logic should go to the parent scope. As it's not the serializer responsibility to know what to serialize